docs/installation/integrations/next-js.md

```diff
@@ -0,0 +1,204 @@
+---
+menu-title: Next.js
+meta-title: Compatibility with Next.js | CKEditor 5 documentation
+meta-description: Learn how to integrate the Next.js framework with the rich text editor - CKEditor 5 - using both the App Router and Pages Router routing strategies.
+category: integrations
+order: 100
+modified_at: 2023-11-14
+---
+
+# Compatibility with Next.js
+
+Next.js is a React meta-framework that helps create full-stack web apps for production. It offers different rendering strategies like server-side rendering (SSR), client-side rendering (CSR), or static site generation (SSG). Additionally, it provides file-based routing, automatic code splitting, and other handy features out of the box.
+
+Next.js 13 introduced a new App Router as an alternative to the previous Pages Router. App Router supports server components and is more server-centric than Pages Router, which is client-side oriented.
+
+CKEditor&nbsp;5 does not support server-side rendering yet, but it is possible to integrate it with the Next.js framework. In this guide, we will add the editor to a Next.js project using both routing paradigms. For this purpose, we will need [Next.js CLI](https://nextjs.org/docs/pages/api-reference/create-next-app), [CKEditor&nbsp;5 online builder](https://ckeditor.com/ckeditor-5/online-builder/), and the official {CKEditor&nbsp;5 @link installation/integrations/react React component}.
+
+## Preparing a build
+
+First, we will use the [online builder](https://ckeditor.com/ckeditor-5/online-builder/). It is a web UI that lets you create a custom build of CKEditor&nbsp;5 and download the code as a zip package.
+
+The online builder is a powerful tool that lets you effortlessly create a rich text editor tailored to your needs. With the online builder, you can choose the desired editor type and plugins, configure the toolbar, and choose the UI language for your editor.
+
+You can learn more about creating custom builds of CKEditor&nbsp;5 with the online builder in the {@link installation/getting-started/quick-start-other#creating-custom-builds-with-online-builder Customized installation} guide.
+
+## Setting up the project
+
+This guide assumes you will use the official CLI tool for Next.js &ndash; `create-next-app`. Refer to the [Next.js documentation](https://nextjs.org/docs/pages/api-reference/create-next-app) to learn how to set up your project.
+
+The CLI will ask you some questions. Depending on your choices, the folder structure in your project may differ.
+
+```plain
```

Comment:
I'm not sure if this is needed. It may change in the future, and we would need to update it / track it.

Reply:
Yeah, that's always a possibility. The guide will be fine without this snippet, as I'm looking at it.